keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -323,6 +333,113 @@ myOtherField: newly added field
 The feature proposed in this KEP would allow this change, since `myField` was
 unaltered from the original object. If `myField` was changed, the new value would have to pass the new validation rule to be written.
 
+### Nested Value Validations 
+
+OpenAPI-schemas support nested validations with logical operations:
+- not
+- oneOf
+- anyOf
+- allOf
+
+To keep the semantics simple and implementation clear, schemas and validations
+nested within the following will be not ratcheted:
+
+- not
+- oneOf
+
+Our intuition is that in the wild CRD authors do not make much use of them, and
+when they do the logic is complicated and references multiple fields. 
+
+For all these constructs, the conservative `DeepEqual` rule still applies 
+to the final output of the rule. 
+
+The following will be ratcheted with the conservative ratcheting rule, and
+their children will also be ratcheted.
+
+- allOf
+- anyOf
+
+### Atomic Lists and Maps
+
+Following Kubernets convention that fields annotated `x-kubernetes-list-type` or `x-kubernetes-map-type` as `atomic` change as a unit, the ratcheting rules
+do not apply to values nested inside atomic lists and maps. But a DeepEqual 
+comparison on the entire object can still ratchet errors for subfields.
+
+### Weaker Ratcheting Rule for Value Validations
+
+Above we propose a conservative catch-all ratcheting rule that should provide
+a minimum level of ratcheting support for any validation rule in the schema.
+
+For most types of validations, the `DeepEqual(OldObject, NewObject)` rule is
+strong enough, but for others it is too weak. Consider the `required` error. This
```

Comment:
strong and weak here is confusing. I know what you mean. I would use the opposite wording: deepequal on the containing object is a very strong requirement. Maybe "strong/weak requirement" makes it clear rather than "rule is strong/weak".

Reply:
Changed to 

> `DeepEqual(OldObject, NewObject)` requirement is very strong. But for some validation rules it is too strict.